uint8 style
int8 position_index
bool predicted
